src/backups.py

```diff
@@ -474,16 +474,14 @@ def _on_s3_credential_changed(self, event: CredentialsChangedEvent):
 
     def _on_create_backup_action(self, event) -> None:  # noqa: C901
         """Request that pgBackRest creates a backup."""
-        backup_type = event.params.get("type", "full").lower()[:4]
-        if backup_type not in ["full", "diff"]:
-            error_message = (
-                f"Invalid backup type: {backup_type}. Possible values: full, differential."
-            )
+        backup_type = event.params.get("type", "full")
+        if backup_type not in BACKUP_TYPE_OVERRIDES:
+            error_message = f"Invalid backup type: {backup_type}. Possible values: {', '.join(BACKUP_TYPE_OVERRIDES.keys())}."
             logger.error(f"Backup failed: {error_message}")
             event.fail(error_message)
             return
 
-        logger.info(f"A {backup_type} backup  has been requested on unit")
+        logger.info(f"A {backup_type} backup has been requested on unit")
```

Comment:
Is it expected to create a full backup if there is no previous backup and the user requests an incremental backup?

Reply:
AFAIU this behavior comes from pgbackrest: It checks internally if it's possible to create a differential/incremental backup; if it's not possible (for any reason), it creates a full backup.
I suppose we could implement this check in the charm ourselves and return error to the user if not possible, but IMO this wouldn't be an easy check to get right, so that's why I left it up to pgbackrest. LMK what's the preferred approach here.

Reply:
Let's follow the default pgbackrest behaviour here. K.I.S.S.
Please re-confirm it with Mohamed in spec.

Reply:
If Mohamed says we should do things differently regarding this behavior, it will probably deserve a separate Jira ticket/PR for the change, as it would be out of scope of this one (which adds incremental option)